integration/v3_lease_test.go

```diff
@@ -233,6 +234,106 @@ func TestV3LeaseExists(t *testing.T) {
 	}
 }
 
+// TestV3LeaseRenewStress keeps creating lease and renewing it immediately to ensure the renewal goes through.
+// it was oberserved that the immediate lease renewal after granting a lease from follower resulted lease not found.
+// related issue https://github.com/coreos/etcd/issues/6978
+func TestV3LeaseRenewStress(t *testing.T) {
+	defer testutil.AfterTest(t)
+	clus := NewClusterV3(t, &ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	cli1 := clus.Client(0)
+	cli2 := clus.Client(1)
+	cli3 := clus.Client(2)
+
+	errc := make(chan error)
+
+	for i := 0; i < 10; i++ {
+		go func() {
+			errc <- stressLeaseRenew(ctx, cli1)
+		}()
+		go func() {
+			errc <- stressLeaseRenew(ctx, cli2)
+		}()
+		go func() {
+			errc <- stressLeaseRenew(ctx, cli3)
+		}()
+	}
+
+	for i := 0; i < 30; i++ {
+		if err := <-errc; err != nil {
+			cancel()
+			t.Fatal(err)
+		}
+	}
+}
+
+func stressLeaseRenew(ctx context.Context, cli *clientv3.Client) error {
+	for ctx.Err() == nil {
+		resp, gerr := cli.Grant(context.TODO(), 60)
+		if gerr != nil {
+			return gerr
+		}
+		_, kerr := cli.KeepAliveOnce(context.TODO(), resp.ID)
+		if kerr != nil {
+			return kerr
+		}
+	}
+	return nil
+}
+
+// TestV3LeaseTimeToLiveStress keeps creating lease and retriving it immediately to ensure the lease can be retrived.
+// it was oberserved that the immediate lease retrival after granting a lease from follower resulted lease not found.
+// related issue https://github.com/coreos/etcd/issues/6978
+func TestV3LeaseTimeToLiveStress(t *testing.T) {
```

Comment:
this is nearly the same thing as the other test, the only difference is whether it calls `stressLeaseTimeToLive` or `stressLeaseRenew`; this can be simplified into a generic test `func testLeaseStress(stresser func(context.Context, leaseClient))` which would be called by both `TestV3Lease` functions